sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/AggregateDocumentQueryExecutionContext.java

```diff
@@ -114,18 +103,49 @@ public AggregateDocumentQueryExecutionContext (IDocumentQueryExecutionComponent<
                 }).flux();
     }
 
-    public static <T extends Resource>  Flux<IDocumentQueryExecutionComponent<T>> createAsync(
-            Function<String, Flux<IDocumentQueryExecutionComponent<T>>> createSourceComponentFunction,
-            Collection<AggregateOperator> aggregates,
-            String continuationToken) {
+    public static <T extends Resource> Flux<IDocumentQueryExecutionComponent<T>> createAsync(
+        Function<String, Flux<IDocumentQueryExecutionComponent<T>>> createSourceComponentFunction,
+        Collection<AggregateOperator> aggregates,
+        Map<String, AggregateOperator> groupByAliasToAggregateType,
+        List<String> groupByAliases,
+        boolean hasSelectValue,
+        String continuationToken) {
 
         return createSourceComponentFunction
-                .apply(continuationToken)
-                .map( component -> { return new AggregateDocumentQueryExecutionContext<T>(component, aggregates);});
+                   .apply(continuationToken)
+                   .map(component -> {
+                       return new AggregateDocumentQueryExecutionContext<T>(component,
+                                                                            new ArrayList<>(aggregates),
+                                                                            groupByAliasToAggregateType,
+                                                                            groupByAliases,
+                                                                            hasSelectValue,
+                                                                            continuationToken);
+                   });
     }
 
     public IDocumentQueryExecutionComponent<T> getComponent() {
         return this.component;
     }
 
+    class RewrittenAggregateProjections {
+        private Document payload;
+
+        public RewrittenAggregateProjections(boolean isValueAggregateQuery, Document document) {
+            if (document == null) {
+                throw new IllegalArgumentException("document cannot be null");
+            }
+            if (isValueAggregateQuery) {
+                this.payload = new Document(document.getPropertyBag());
+            } else {
+                if (document.get("payload") instanceof ObjectNode) {
```

Comment:
if "payload" does not exist, then this should throw an exception.

Reply:
Added check